libs/driver/src/VideoDriver.cpp

```diff
@@ -73,4 +76,60 @@ void VideoDriver::SetNewSize(VideoMode windowSize, Extent renderSize)
 {
     windowSize_ = windowSize;
     renderSize_ = renderSize;
+
+    const auto ratioXY = PointF(renderSize_) / PointF(windowSize_.width, windowSize_.height);
+    dpiScale_ = (ratioXY.x + ratioXY.y) / 2.f; // use the average ratio of both axes
+
+    if(autoGuiScale_)
+    {
+        unsigned percent;
+        getGuiScaleRange(nullptr, nullptr, &percent);
+        guiScale_ = GuiScale(percent);
```

Comment:
Triggered by the codecov annotation IMO it makes sense to add a short test case explaining & showing what effect `autoGuiScale_` is supposed to have. Is that easily possible?

Reply:
Can do.